### ISSUE TITLE
How can I get values from an object being the dynamic key?

### ISSUE THREAD

**Issue by emtudo** [rayepps/radash#91]
example:

function getByKey(item, key) { // this is example
   return get(item, key) // lodash
}

const object= {
   'a': { b: { c: 'test' } }
}

getKeyKey('a.b.c') // returns test

How can I do the same with radash?

**Comment by matteosacchetto** [comment]
Hi there,
so, it is not using radash, but it is in plain JS.

Here is an implementation, maybe not the best one, but it does the job
```js
const assert = (cond, msg) => {
    if(!cond) {
        throw new Error(msg)
    }
}

const getByKey = (item, key) => {
    assert(/^(\w+\??\.)*\w+$/g.test(key), `The provided key: ${key} is malformed`)
    return key
        .split('.')
        .reduce((o, k) => {
            return k.at(-1) === '?' ? o[k.slice(0, -1)] ?? {} : o[k]
        }, item)
}

const obj = {
    foo: {
        b: true
    }
}
console.log(getByKey(obj, 'test?.a')); // undefined
console.log(obj.test?.a); // undefined

console.log(getByKey(obj, 'foo?.a')); // undefined
console.log(obj.foo?.a); // undefined

console.log(getByKey(obj, 'foo?.b')); // true
console.log(obj.foo?.b); // true
```

It should behave in the same way as the `?.` operator in modern JS

Cheers

**Comment by matteosacchetto** [comment]
```js
const getByKey(item, key) => {
  return key
    .split(/(\w+)?(?:\[(\w+)\]|\['(\w+)'\]|\["(\w+)?"\])?(\?)?(?:\.)?/)
    .filter((el) => el)
    .reduce((o, k) => {
      return k === '?' ? o ?? {} : o[k];
    }, item);
}
console.log(getByKey(obj, 'foo["b"]')); // true
console.log(obj.foo['b']); // true
```

This should be a more advanced version supporting also the array notation
Again, maybe not the best implementation, but it should do the job

**Comment by rayepps** [comment]
Hey @emtudo a PR was just merged to support this (#57) so the `get` function now works by path.

```ts
import { get } from 'radash'
const person = {
    attributes: {
        age: {
            years: 22
        }
    }
}

get(person, 'attributes.age.years') // => 22
```

**Comment by emtudo** [comment]
thanks!
